pkg/docker/storage.jsx

```diff
@@ -0,0 +1,440 @@
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/react",
+    "base1/cockpit",
+    "performance/dialog-view"
+], function($, React, cockpit, dialog_view) {
+    var _ = cockpit.gettext;
+    var C_ = cockpit.gettext;
+
+    function init_model(callback) {
+        var pool_devices = [ ];
+
+        var self = {
+            error: null,
+            pool_devices: [ ],
+            extra_devices: [ ],
+            total: 0,
+            used: 0
+        };
+
+        cockpit.spawn([ "/usr/libexec/cockpit-atomic-storage-monitor" ],
+                      { superuser: true }).
+                stream(function (data) {
+                    // XXX - find the newlines here
+                    console.log("INFO", data);
+                    var info = JSON.parse(data);
+                    self.pool_devices = info.pool_devices;
+                    self.extra_devices = info.extra_devices;
+                    self.total = info.total;
+                    self.used = info.used;
+                    $(self).triggerHandler("changed");
+                }).
+                fail(function (error) {
+                    console.warn(error);
+                    self.error = error.problem || "broken";
+                    $(self).triggerHandler("changed");
+                });
+        callback(self);
+    }
+
+    var model_promise = null;
+
+    function get_storage_model() {
+        if (model_promise)
+            return model_promise;
+
+        var dfd = $.Deferred();
+        init_model(function (model) {
+            dfd.resolve(model);
+        });
+
+        model_promise = dfd.promise();
+        return model_promise;
+    }
+
+    var DriveBox = React.createClass({
+        getInitialState: function () {
+            return {
+                drives: this.props.model.extra_devices,
+                checked: { }
+            };
+        },
+        componentDidMount: function () {
+            var self = this;
+            $(this.props.model).on("changed", function () {
+                self.setState({ drives: self.props.model.extra_devices });
+            });
+        },
+
+        driveChecked: function (drive) {
+            return !!this.state.checked[drive.path];
+        },
+        makeDriveToggler: function (drive) {
```

Comment:
instead of returning a function, you could take `event` and `drive` as a parameter and bind the function.

Example: https://github.com/cockpit-project/cockpit/blob/master/pkg/base1/cockpit-components-listing.jsx#L162